### PR TITLE
Fix Azure Image Standard test for FreeBSD

### PR DESCRIPTION
**Pull request by dsrivastavv**
Tests fixed:
- `verify_resource_disk_file_system`: Fixed
- `verify_resource_disk_readme_file`: Fixed
- `verify_boot_error_fail_warnings` : Can't fix as the error seems legit. Error found: `error: Fssh_kex_exchange_identification: Connection closed by remote host`
- `verify_grub`: Skipped

### CODE REVIEW

File: lisa/tools/docker.py [hunk 1]
```diff
@@ -137,6 +137,13 @@ def _install(self) -> bool:
             self.node.os.install_packages(["moby-engine", "moby-cli"])
         elif isinstance(self.node.os, SLES):
             self.node.os.install_packages(["docker"])
+        elif isinstance(self.node.os, BSD):
```
Comment (squirrelsc) [inline lisa/tools/docker.py]: Should it be in a separated PR?

File: microsoft/testsuites/core/azure_image_standard.py [hunk 2]
```diff
@@ -992,9 +993,10 @@ def verify_resource_disk_readme_file(self, node: RemoteNode) -> None:
         node.features[Disk].get_partition_with_mount_point(resource_disk_mount_point)
 
         # verify lost+found folder exists
-        fold_path = f"{resource_disk_mount_point}/lost+found"
-        folder_exists = node.tools[Ls].path_exists(fold_path, sudo=True)
-        assert_that(folder_exists, f"{fold_path} should be present").is_true()
+        if not isinstance(node.os, BSD):
```
Comment (squirrelsc) [inline microsoft/testsuites/core/azure_image_standard.py:1000]: Add comments above to explain why FreeBSD skip this check.

File: lisa/tools/lsblk.py [hunk 3]
```diff
@@ -65,6 +65,10 @@ class DiskInfo(object):
     type: str = ""
     partitions: List[PartitionInfo] = field(default_factory=list)
 
+    @classmethod
+    def _freebsd_tool(cls) -> Optional[Type[Tool]]:
```
Comment (squirrelsc) [inline lisa/tools/lsblk.py]: Why does the disk info need this method?

File: lisa/tools/lsblk.py [hunk 4]
```diff
@@ -228,3 +234,109 @@ def find_mountpoint_by_volume_name(
...
+                    dev_type=entry["type"],
+                    partitions=disk_partition_map.get(entry["name"], []),
+                )
+            )
+
+        # sort disk with OS disk first
+        disks.sort(key=lambda disk: disk.is_os_disk, reverse=True)
+        return disks
+
+    @staticmethod
```
Comment (squirrelsc) [inline lisa/tools/lsblk.py]: This method doesn't need to be static.

Comment (dsrivastavv) [reply lisa/tools/lsblk.py]: We are not using any class information. We can also move it to common function if you feel so

File: lisa/tools/docker.py (hunk 1 above)
Comment (dsrivastavv) [reply lisa/tools/docker.py]: Rebased to main

File: lisa/tools/lsblk.py (hunk 3 above)
Comment (dsrivastavv) [reply lisa/tools/lsblk.py]: This was an error. Fixed

File: lisa/tools/lsblk.py (hunk 4 above)
Comment (squirrelsc) [reply lisa/tools/lsblk.py]: You can move it to a function in the same file. It's not shared by other classes, so it doesn't need to move to common.

**Comment by dsrivastavv** [comment]
@squirrelsc Not skipping `verify_repository_installed` testcase until Wei can confirm. But this doesn't need to block this PR

File: lisa/tools/lsblk.py [hunk 5]
```diff
@@ -3,13 +3,24 @@
...
-from typing import Dict, List, Optional
+from typing import Dict, List, Optional, Type
 
 from lisa.executable import Tool
-from lisa.operating_system import FreeBSD, Posix
+from lisa.operating_system import BSD, Posix
 from lisa.util import LisaException, find_patterns_groups_in_lines
 
 
+def get_size_in_bytes(size: int, size_unit: str) -> int:
```
Comment (squirrelsc) [inline lisa/tools/lsblk.py]: Because it's not used out of this file, still make it preview starts with underscore.

Comment (dsrivastavv) [reply lisa/tools/lsblk.py]: fixed

**Review by squirrelsc — APPROVED** [review]

**Pull request merged**